### ISSUE TITLE
Unable to handle .js.erb file?

### ISSUE THREAD

**Issue by mngb** [rails/propshaft#214]
The previous pipe-line tool sprockets-rails can pack .js.erb to .js asset.
Does propshaft can do this?

**Comment by brenogazzola** [comment]
No. Propshaft has a much smaller scope than Sprockets and only handles fingerprinting your assets and serving them. 

If you are using js.erb because you needed to reference assets in your JavaScript files, please check the recently added macro in the README. 

https://github.com/rails/propshaft?tab=readme-ov-file#referencing-digested-assets-in-css-and-javascript

**Comment by mngb** [comment]
Thanks for you reply.
I'm using .js.erb file to share some constant value between js and ruby file, e.g ErrorCode defined in ruby.
I found the .erb was packed to asset folder, does any post hook can support to implement this?

**Comment by brenogazzola** [comment]
There are two ways I can think that might work. The first one is to enhance the precompile task, like I mentioned in another issue:

https://github.com/rails/propshaft/issues/158#issuecomment-1786039189

The other is to rename your files to plain JS files and create your own compiler use the JsAssetUrl as an example:

https://github.com/rails/propshaft/blob/main/lib/propshaft/compiler/js_asset_urls.rb

**Comment by rmacklin** [comment]
If all you need is to "share some constant value between js and ruby", I'd suggest a different approach. With a small refactor, you can pass the values to javascript through HTML, either as data-attributes, meta tag values, or an inline `script` element.

In addition to removing the need for a js.erb compilation step, this means the JS assets can remain cached when only those ruby values change.

**Comment by brenogazzola** [comment]
I’ll close this issue since it’s not really a bug and @rmacklin and I gave a few suggestions on how to handle it. That said, feel free to keep commenting.
